### PR TITLE
Remove astropy_helpers and update packaging to use recommendations in APE 17

### PR DESCRIPTION
**Pull request by Cadair**
This PR contains a few packaging changes to follow the work being done in [APE 17](https://github.com/astropy/astropy-APEs/pull/52)

The only noticeable change here is that `python setup.py test` and `python setup.py build_docs` are no longer supported. If you still run those commands the alternatives are printed, with links to the documentation. 

I will hold off on merging this, so that a) maintainers see it and know about the changes to the setup commands and b) wait until extension helpers has a chance to mature a little.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -5,6 +5,7 @@ requires = [
          "wheel",
          "numpy==1.14.5; python_version<='3.7'",
          "numpy==1.17.3; python_version>='3.8'",
```
Comment (astrofrog) [inline pyproject.toml]: Could try using ``oldest-compatible-numpy``?

**Comment by Cadair** [comment]
This is an interesting meta question about the changelog. Is the changelog just for users or also for developers? Personally, I think this is an infrastructure change which shouldn't go in the changelog.

I will be sending info about this around to `sunpy-dev` before we merge it though.

**Comment by dstansby** [comment]
Isn't there a `dev` or similar category in the changelog types?

**Comment by Cadair** [comment]
There is an internal one, I suppose it could go there. I still think that there is a policy decision to be made here (but this probably isn't the place).

**Comment by ayshih** [comment]
I'd like to see `make html` mentioned as an option for building the docs (speaking as someone who keeps forgetting whether `tox` actually works on his setup).

I support adding a changelog entry for this.  There may be some users using the `python setup.py ...` commands, e.g., to make sure that their setup isn't borked or to have docs available offline.

File: tox.ini [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{36,37,38}-{offline,online,astropydev,numpydev,astropy31},build_docs,figure,figure_astropydev,conda
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply tox.ini]: Oh I see. you can remove them from the list but still call the directly.

**Comment by nabobalis** [comment]
I added a changelog and added astropyfrog's suggestion.

**Comment by nabobalis** [comment]
Do we want to backport this to 1.1?

File: tox.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (Cadair) [reply tox.ini]: Yeah it means that you can run `tox` and get a minimal set locally.

**Comment by Cadair** [comment]
The main thing I want to wait for here is a non-pre release of extension-helpers.

**Comment by nabobalis** [comment]
Seems an API change from extension_helpers came.

File: tox.ini [hunk 3]
```diff
@@ -27,7 +26,7 @@ deps =
     astropydev,build_docs: astroquery
     astropydev,build_docs: jplephem
 commands =
-    offline,astropydev,numpydev,astropy31: {env:PYTEST_COMMAND} {posargs}
```
Comment (Cadair) [inline tox.ini]: @nabobalis did you drop the numpydev build?

Comment (nabobalis) [reply tox.ini]: I did

**Comment by Cadair** [comment]
Given the release of extension helpers, I think this is good for merging once approved.

**Pull request merged**